src/main/java/frc/robot/subsystems/ElevatorSubsystem.java

```diff
@@ -0,0 +1,147 @@
+    // SmartDashboard.putData("Elevator Sim", m_mech2d);
+  }
+
+  public Command sysIdQuasistatic(SysIdRoutine.Direction direction) {
+    return routine.quasistatic(direction);
+  }
+
+  public Command sysIdDynamic(SysIdRoutine.Direction direction) {
+    return routine.dynamic(direction);
+  }
+
+  public void voltageDrive(Voltage drive) {
+    m_motor.setVoltage(drive.in(Units.Volts));
+    m_motor2.setVoltage(drive.in(Units.Volts));
+  }
+
+  public void logMotors(SysIdRoutineLog log) { // in theory this should work?
+    log.motor("elevator-motor")
+        .voltage(
+            m_appliedVoltage.mut_replace(
+                m_motor.get() * RobotController.getBatteryVoltage(), Units.Volts))
+        .angularPosition(m_motor.getPosition().getValue())
+        .angularVelocity(m_motor.getVelocity().getValue());
+    log.motor("elevator-motor2")
+        .voltage(
+            m_appliedVoltage.mut_replace(
+                m_motor2.get() * RobotController.getBatteryVoltage(), Units.Volts))
+        .angularPosition(m_motor2.getPosition().getValue())
+        .angularVelocity(m_motor2.getVelocity().getValue());
+  }
+
+  public void reachGoal(double goal) {
+    m_controller.setGoal(goal);
+    // With the setpoint value we run PID control like normal
+    double pidOutput = m_controller.calculate(m_motor.getPosition().getValueAsDouble());
+    double feedforwardOutput =
+        0; // m_feedforward.calculate(m_controller.getSetpoint().velocity); can be used later to
+    // calculate with correct values
+    m_motor.setVoltage(pidOutput + feedforwardOutput);
+  }
+
+  public void motorConfigs() {
+    var talonFXConfigurator = m_motor.getConfigurator();
+    var talonFXConfigurator2 = m_motor2.getConfigurator();
+    var currentLimits = new CurrentLimitsConfigs();
+    var softLimits = new SoftwareLimitSwitchConfigs();
+    // soft limits
+    softLimits.ForwardSoftLimitEnable = true;
+    softLimits.ReverseSoftLimitEnable = true;
+    softLimits.ForwardSoftLimitThreshold = forwardSoftLimit;
+    softLimits.ReverseSoftLimitThreshold = reverseSoftLimit;
+    talonFXConfigurator.apply(softLimits);
+    talonFXConfigurator2.apply(softLimits);
+    // enable stator current limit
+    currentLimits.StatorCurrentLimit = 5; // starting low for testing
+    currentLimits.StatorCurrentLimitEnable = true;
+    currentLimits.SupplyCurrentLimit = 5; // starting low for testing
+    currentLimits.SupplyCurrentLimitEnable = true;
+    talonFXConfigurator.apply(currentLimits);
+    talonFXConfigurator2.apply(currentLimits);
+    // create brake mode for motors
+    var outputConfigs = new MotorOutputConfigs();
+    outputConfigs.NeutralMode = NeutralModeValue.Brake;
+    talonFXConfigurator.apply(outputConfigs);
+    talonFXConfigurator2.apply(outputConfigs);
+  }
+
+  public Command goUp() {
+    return startEnd(
+        () -> m_motor.set(-0.25), () -> m_motor.set(-0.02)); // test values from dev model bot
+  }
```

Comment:
Would be best to define `-0.25` and `-0.02` above as constant variables & use those. Something like UP_VOLTAGE, DOWN_VOLTAGE and HOLD_VOLTAGE